docs/index.rst

```diff
@@ -17,68 +40,138 @@ Read a beam from a fits header::
+
+    >>> my_beam_ellip = Beam(major=0.5*u.arcsec, minor=0.25*u.arcsec, pa=30*u.deg)
+    >>> my_beam_ellip
+    Beam: BMAJ=0.5 arcsec BMIN=0.25 arcsec BPA=30.0 deg
+
+The beam area in steradians is::
 
+    >>> my_beam_ellip.sr # doctest: +FLOAT_CMP
+    <Quantity 3.3290795e-12 sr>
 
-Use a beam for Jy -> K conversion::
+Or projected into physical units given a distance::
+
+    >>> my_beam_ellip.beam_projected_area(840*u.kpc).to(u.pc**2) # doctest: +FLOAT_CMP
+    <Quantity 2.3489985 pc2>
+
+A common unit conversion in radio astronomy is Jy/beam to K, which depends on the beam area.
+A `~radio_beam.Beam` object can be used for this unit conversion with `~astropy.units`::
 
     >>> (1*u.Jy).to(u.K, u.brightness_temperature(25*u.GHz, my_beam)) # doctest: +FLOAT_CMP
     <Quantity 7821.572919292681 K>
 
-Convolve with another beam::
+Or alternatively with::
+
+    >>> (1*u.Jy).to(u.K, my_beam.jtok_equiv(25*u.GHz)) # doctest: +FLOAT_CMP
+    <Quantity 7821.572919292681 K>
+
+To get the value off 1 Jy in K for a given beam::
+
+    >>> my_beam.jtok(25*u.GHz) # doctest: +FLOAT_CMP
+    <Quantity 7821.572919292681 K>
+
+Two beams can be convolved::
 
     >>> my_asymmetric_beam = Beam(0.75*u.arcsec, 0.25*u.arcsec, 0*u.deg)
     >>> my_other_asymmetric_beam = Beam(0.75*u.arcsec, 0.25*u.arcsec, 90*u.deg)
-    >>> my_asymmetric_beam.convolve(my_other_asymmetric_beam)  # doctest: +SKIP
+    >>> my_asymmetric_beam.convolve(my_other_asymmetric_beam) # doctest: +FLOAT_CMP
     Beam: BMAJ=0.790569415042 arcsec BMIN=0.790569415042 arcsec BPA=45.0 deg
 
-Deconvolve another beam::
+And also deconvolved::
 
     >>> my_big_beam = Beam(1.0*u.arcsec, 1.0*u.arcsec, 0*u.deg)
     >>> my_little_beam = Beam(0.5*u.arcsec, 0.5*u.arcsec, 0*u.deg)
-    >>> my_big_beam.deconvolve(my_little_beam)  # doctest: +SKIP
+    >>> my_big_beam.deconvolve(my_little_beam) # doctest: +FLOAT_CMP
     Beam: BMAJ=0.866025403784 arcsec BMIN=0.866025403784 arcsec BPA=0.0 deg
 
-Read a table of beams::
+An error is returned if the beam area is too small to deconvolve from the other::
+
+    >>> my_little_beam.deconvolve(my_big_beam)  # doctest: +SKIP
+
+To find the smallest common beam between any two beams::
+
+    >>> my_asymmetric_beam.commonbeam_with(my_other_asymmetric_beam) # doctest: +FLOAT_CMP
+    Beam: BMAJ=0.75 arcsec BMIN=0.75 arcsec BPA=90.0 deg
+
+Handling a sets of beams
+------------------------
+
+`~radio_beam.Beams` handles operations on sets of beams.
+
+To read a table of beams from a FITS table::
 
     >>> from radio_beam import Beams
     >>> from astropy.io import fits
     >>> bin_hdu = fits.open('file.fits')[1]  # doctest: +SKIP
     >>> beams = Beams.from_fits_bintable(bin_hdu)  # doctest: +SKIP
 
+In the above example, the second FITS extension contains the beam tables, while the
+first would have the spectral cube data.
+
+To read a table of beams from a CASA image (must be run inside a CASA environment!)::
```

Comment:
can we do this w/casa-formats-io now?

Reply:
Probably. I raised an issue: https://github.com/radio-astro-tools/casa-formats-io/issues/42.